### PR TITLE
feat: add clawback cli and tests

### PR DESCRIPTION
**Pull request by mattverse**
## Description

This PR adds cli for clawback vesting account, tests and test suites for the added cli as well.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/auth/vesting/client/testutil/suite.go [hunk 1]
```diff
@@ -130,3 +130,233 @@ func (s *IntegrationTestSuite) TestNewMsgCreateVestingAccountCmd() {
...
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			},
+			expectErr:    false,
+			expectedCode: 0,
+			respType:     &sdk.TxResponse{},
+		},
+	} {
```
Comment (alexanderbez) [inline x/auth/vesting/client/testutil/suite.go:346]: Why is the test execution block wrapped in parentheses?

File: x/auth/vesting/client/testutil/suite.go [hunk 2]
```diff
@@ -130,3 +130,233 @@ func (s *IntegrationTestSuite) TestNewMsgCreateVestingAccountCmd() {
...
+		},
+		{
+			name: "bad vesting periods",
+			args: []string{
+				sdk.AccAddress("addr13______________").String(),
+				fmt.Sprintf("--%s=%s", cli.FlagVesting, "testdata/badperiod.json"),
+			},
+			expectErr: true,
+		},
+	} {
```
Comment (alexanderbez) [inline x/auth/vesting/client/testutil/suite.go:264]: ditto, Why is the test execution block wrapped in parentheses?

File: x/auth/vesting/client/cli/tx.go [hunk 3]
```diff
@@ -79,3 +86,167 @@ timestamp.`,
...
+		if err != nil {
+			return 0, nil, err
+		}
+		if p.Length < 1 {
+			return 0, nil, fmt.Errorf("invalid period length of %d in period %d, length must be greater than 0", p.Length, i)
+		}
+		period := types.Period{Length: p.Length, Amount: amount}
+		periods = append(periods, period)
+	}
+	return startTime, periods, nil
```
Comment (alexanderbez) [inline x/auth/vesting/client/cli/tx.go]:
```suggestion
	contents, err := ioutil.ReadFile(path)
	if err != nil {
		return 0, nil, err
	}

	var data VestingData
	if err := json.Unmarshal(contents, &data); err != nil {
		return 0, nil, err
	}

	startTime := data.StartTime
	periods := make([]types.Period, len(data.periods))

	for i, p := range data.Periods {
		amount, err := sdk.ParseCoinsNormalized(p.Coins)
		if err != nil {
			return 0, nil, err
		}
		if p.Length < 1 {
			return 0, nil, fmt.Errorf("invalid period length of %d in period %d, length must be greater than 0", p.Length, i)
		}

		periods[i] = types.Period{Length: p.Length, Amount: amount}
	}

	return startTime, periods, nil
```

File: x/auth/vesting/client/cli/tx.go [hunk 4]
```diff
@@ -79,3 +86,167 @@ timestamp.`,
...
+If both files are given, they must describe schedules for the same total amount.
+If one file is omitted, it will default to a schedule that immediately unlocks or vests the entire amount.
+The described amount of coins will be transferred from the --from address to the vesting account.
+Unvested coins may be "clawed back" by the funder with the clawback command.
+Coins may not be transferred out of the account if they are locked or unvested, but may be staked.
+Staking rewards are subject to a proportional vesting encumbrance.
+
+A periods file is a JSON object describing a sequence of unlocking or vesting events,
+with a start time and an array of coins strings and durations relative to the start or previous event.`,
+		Example: `Sample period file contents:
```
Comment (alexanderbez) [inline x/auth/vesting/client/cli/tx.go:149]: woaaa, `Example` was always part of the `cobra.Command`? Neat!

File: x/auth/vesting/client/cli/tx.go [hunk 5]
```diff
@@ -79,3 +86,167 @@ timestamp.`,
...
+			  {
+				  "coins": "10test",
+				  "length": 2592000 //30 days
+			  },
+			  {
+				"coins": "10test",
+				"length": 2592000 //30 days
+			}
+		]}
+	    `,
```
Comment (alexanderbez) [inline x/auth/vesting/client/cli/tx.go]:
```suggestion
{
  "start_time": 1625204910,
  "periods": [
    {
      "coins": "10test",
      "length": 2592000 // 30 days
    },
    {
      "coins": "10test",
      "length": 2592000 // 30 days
    }
  ]
}
`,
```

File: x/auth/vesting/client/cli/tx.go [hunk 6]
```diff
@@ -79,3 +86,167 @@ timestamp.`,
...
+				return err
+			}
+
+			lockupFile, _ := cmd.Flags().GetString(FlagLockup)
+			vestingFile, _ := cmd.Flags().GetString(FlagVesting)
+			if lockupFile == "" && vestingFile == "" {
+				return fmt.Errorf("must specify at least one of %s or %s", FlagLockup, FlagVesting)
+			}
+			var lockupStart, vestingStart int64
+			var lockupPeriods, vestingPeriods []types.Period
```
Comment (alexanderbez) [inline x/auth/vesting/client/cli/tx.go]:
```suggestion
			}

			var (
			  lockupStart, vestingStart int64
			  lockupPeriods, vestingPeriods []types.Period
			)
```

File: x/auth/vesting/client/cli/tx.go [hunk 7]
```diff
@@ -79,3 +86,167 @@ timestamp.`,
...
+			var dest sdk.AccAddress
+			destString, _ := cmd.Flags().GetString(FlagDest)
+			if destString != "" {
+				dest, err = sdk.AccAddressFromBech32(destString)
+				if err != nil {
+					return fmt.Errorf("bad dest address: %w", err)
+				}
+			}
+
+			msg := types.NewMsgClawback(clientCtx.GetFromAddress(), addr, dest)
```
Comment (alexanderbez) [inline x/auth/vesting/client/cli/tx.go:250]: If `destString` is empty, then `dest` never gets populated. Is that OK or acceptable?

**Review by alexanderbez — COMMENTED** [review]
LGTM. Just left some minor feedback :)

File: x/auth/vesting/client/testutil/suite.go (hunk 1 above)
Comment (mattverse) [reply x/auth/vesting/client/testutil/suite.go:346]: `for` is wrapped around testcases itself!

File: x/auth/vesting/client/testutil/suite.go (hunk 2 above)
Comment (mattverse) [reply x/auth/vesting/client/testutil/suite.go:264]: ditto, for is wrapped around testcases itself!

File: x/auth/vesting/client/cli/tx.go (hunk 4 above)
Comment (mattverse) [reply x/auth/vesting/client/cli/tx.go:149]: We should add this to gamm create-pool as well!

File: x/auth/vesting/client/cli/tx.go (hunk 7 above)
Comment (mattverse) [reply x/auth/vesting/client/cli/tx.go:250]: yeah! It's ok (handled in msg_server.go) good catch!

File: x/auth/vesting/client/cli/cli_test.go [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+	s.cfg = network.DefaultConfig()
+
+	// modification to pay fee with test bond denom "stake"
+	// genesisState := app.ModuleBasics.DefaultGenesis(s.cfg.Codec)
+	// gammGen := gammtypes.DefaultGenesis()
+	// gammGen.Params.PoolCreationFee = sdk.Coins{sdk.NewInt64Coin(s.cfg.BondDenom, 1000000)}
+	// gammGenJson := s.cfg.Codec.MustMarshalJSON(gammGen)
+	// genesisState[gammtypes.ModuleName] = gammGenJson
+	// s.cfg.GenesisState = genesisState
```
Comment (alexanderbez) [inline x/auth/vesting/client/cli/cli_test.go]:
```suggestion
```

**Review by alexanderbez — APPROVED** [review]
🚀

**Pull request merged**